### DISCUSSION TITLE
NFS Mount permission issues

### DISCUSSION THREAD

**Discussion by jmanis2** [jlesage/docker-nginx-proxy-manager#308] [General]
In the issue 228 you asked in the comment if the user was using an NFS mount:
https://github.com/jlesage/docker-nginx-proxy-manager/issues/228#issuecomment-1009260480

I'm having the same issue described and am using an NFS mount. This issue occurs not just on your project but others (but not all others).  Was there a particular remedy or comment you would have provided if the user had confirmed they were using NFS volumes?

Thanks
Jeff

**Comment by jlesage** [comment]
Using NFS mount can be tricky and it's easy to misconfigure the container and/or NFS server.
To determine why the "permission denied" error is showing, we need the NFS export used on the server side along with the container configuration (i.e how it has been created).

**Comment by jmanis2** [comment]
I sorted this out on my end. It was the permission squashing on the NFS server side. In my case I’m using a Synology which squashes the permissions to the Admin user rather than to root. The docker server is running Photon OS which runs the docker service under root. Synology squashing to the admin user caused the problem.  Turning off the permission squashing resolved the problem for this and several other containers where the config directory has to be writable by an early provisioning process of the container (e.g. splunk uses a terraform process to build the container once it boots). 

I’d consider running your NPM on a local volume but there’s no backup/restore functionality in the config so the only way to protect my configuration settings. 

Thanks for offering to help troubleshooting.
